neo/Consensus/ConsensusContext.cs

```diff
@@ -75,6 +78,46 @@ public Block CreateBlock()
             return block;
         }
 
+        public void Deserialize(BinaryReader reader)
+        {
+            if (reader.ReadUInt32() != Version) return;
+            State = (ConsensusState)reader.ReadByte();
+            PrevHash = reader.ReadSerializable<UInt256>();
+            BlockIndex = reader.ReadUInt32();
+            ViewNumber = reader.ReadByte();
+            Validators = reader.ReadSerializableArray<ECPoint>();
+            MyIndex = reader.ReadInt32();
+            PrimaryIndex = reader.ReadUInt32();
+            Timestamp = reader.ReadUInt32();
+            Nonce = reader.ReadUInt64();
+            NextConsensus = reader.ReadSerializable<UInt160>();
+            if (NextConsensus.Equals(UInt160.Zero))
+                NextConsensus = null;
+            TransactionHashes = reader.ReadSerializableArray<UInt256>();
+            if (TransactionHashes.Length == 0)
+                TransactionHashes = null;
+            Transaction[] transactions = new Transaction[reader.ReadVarInt()];
```

Comment:
Where is the corresponding WriteVarInt that this is supposed to be reading for the length of transactions? For the way this it is serialized, should this be just doing reader.ReadSerializableArray<Transaction>()?

Reply:
I wrote tests, and verified it works correctly. I see it is because it is serialized by `Write<T>(this BinaryWriter writer, T[] value) where T : ISerializable` from in IO/Helper.cs which does use `WriteVarInt`.